docs/tools/mbed_cli_2/migration_guide.md

```diff
@@ -0,0 +1,188 @@
+- [Embitz](https://github.com/ARMmbed/mbed-tools/issues/251)
+- [IAR Embedded Workbench for Arm](https://github.com/ARMmbed/mbed-tools/issues/252)
+- [MCUXpresso](https://github.com/ARMmbed/mbed-tools/issues/253)
+- [NetBeans](https://github.com/ARMmbed/mbed-tools/issues/254)
+- [STM32CubeIDE](https://github.com/ARMmbed/mbed-tools/issues/257)
+- [Keil uVision](https://github.com/ARMmbed/mbed-tools/issues/256)
+
+### Library management
+
+<span style="background-color:#ffcccc;">The subcommands `mbed-cli add` (add
+library from URL) and `mbed-cli remove` (remove library) are deprecated.
+Instead, use `git clone` or `mbed-tools import` to clone a library. Then,
+manually create a `reponame.lib` file that contains a single line like
+`https://github.com/ARMmbed/reponame#branch-or-tag`, pointing to your repo at
+the default branch (no `#` and nothing after the `#`), desired tag (append with
+`#tagname`), or desired branch (append with `#branchname`).</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli new --library`
+(force creation of an Mbed library) is deprecated. Create a new Mbed library by
+hand instead, using standard git commands.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli ls` (view
+dependency tree)" is deprecated. The subcommand `mbed-tools deploy` will now
+display all dependencies.</span><br>
+
+### Tool configuration
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli config` (tool
+configuration)" is deprecated. There is no need to configure an Mbed-specific
+compiler path in Mbed CLI 2; the compiler path is determined in a standard way
+by CMake instead. Other configuration options, like "target", "toolchain",
+"protocol", "depth" and "cache", are not supported in Mbed CLI 2.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli toolchain`
+(set or get default toolchain) is deprecated. Pass the desired toolchain on the
+command line when compiling with `mbed-tools compile --toolchain GCC_ARM`, to,
+for example, select `GCC_ARM` as the toolchain.</span><br>
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-cli target` (set
+or get default target)" is deprecated. Pass the desired target on the command
+line when compiling with `mbed-tools compile --mbed-target
+DISCO_L475VG_IOT01A`, to, for example, select `DISCO_L475VG_IOT01A` as the
+Mbed target.</span><br>
+
+
+### Creating an Mbed program
+<span style="background-color:#ffe6cc;">The subcommand `mbed-tools new` creates
+a new Mbed program by default and supports only one command-line-option
+`--create-only` (create a program without fetching mbed-os). The other
+command-line-options supported in Mbed CLI 1, as follow, are deprecated:
+
+* `--scm` (now always git)
+* `--program` (now always a program)
+* `--library` (now always a program)
+* `--mbedlib` (instead, add with a separate subsequent command)
```

Comment:
From the CLI 1 documentation and in:
https://github.com/ARMmbed/mbed-cli/blob/master/mbed/mbed.py#L2151
https://github.com/ARMmbed/mbed-cli/blob/master/mbed/mbed.py#L140 
the `--mbedlib` flag in mbed-cli imports the Mbed 2 library instead of Mbed OS, so this line should explain this is deprecated.